umbral/umbral.py

```diff
@@ -49,13 +76,44 @@ def is_consistent(self, vKeys, params: UmbralParameters):
 
         return lh_exp == rh_exp
 
+    def __bytes__(self):
+        return self.to_bytes()
+
+
 class CapsuleFrag(object):
     def __init__(self, e1, v1, id_, x):
         self.e1 = e1
         self.v1 = v1
         self.bn_kfrag_id = id_
         self.point_eph_ni = x
 
+    @classmethod
+    def from_bytes(cls, data: bytes, curve):
+        """
```

Comment:
I am slightly allergic to `data` as a var name.  What is this actually - cfrag_as_bytes, right?  That seems like a better name.

And then also, I don't love all the int literals as splitter fodder.  This is one of those times where I think we're better off using `ByteStringSplitter` as an separate project.  

For now, the ints are OK, but they trigger my sense of danger a little bit.